frostsnapp/native/src/coordinator.rs

```diff
@@ -161,28 +175,54 @@ impl FfiCoordinator {
                 let mut coordinator_outbox = pending_loop.lock().unwrap();
                 let mut messages_from_devices = vec![];
                 let mut db = db_loop.lock().unwrap();
+                let mut device_list = device_list.lock().unwrap();
 
                 // process new messages from devices
                 {
-                    for change in &device_changes {
+                    for change in device_changes {
+                        device_list.consume_manager_event(change.clone());
                         match change {
                             DeviceChange::Registered { id, .. } => {
                                 if let Some(protocol) = &mut *ui_protocol_loop {
-                                    protocol.connected(*id);
+                                    protocol.connected(id);
+                                }
+
+                                if let Some(connected_device) = device_list.get_device(id) {
+                                    // we only send some messages out if the device has up to date firmware
+                                    if !connected_device.needs_firmware_upgrade().0 {
+                                        coordinator_outbox.extend(
+                                            coordinator.maybe_request_nonce_replenishment(id),
```

Comment:
Does this belong here/in this PR? - nonce replenishment has been in the code for a while, or is it to handle per-coordinator nonces?

Reply:
Yes we shouldn't request nonces from devices unless they have update firmware. We should only send the messages needed to updated firmware.